src/LogicItLab/Salesforce/MapperBundle/WsdlValidator.php

```diff
@@ -0,0 +1,154 @@
+<?php
+
+namespace LogicItLab\Salesforce\MapperBundle;
+
+use LogicItLab\Salesforce\MapperBundle\Annotation\AnnotationReader;
+use Symfony\Component\Finder\Finder;
+
+class WsdlValidator
+{
+    /** @var AnnotationReader */
+    private $annotationReader;
+
+    /** @var string */
+    private $baseProjectDir;
+
+    /** @var string */
+    private $modelDirPath;
+
+    /** @var string */
+    private $wsdlPath;
+
+    /** @var string */
+    private $wsdlContents;
+
+    /**
+     * @param AnnotationReader $annotationReader
+     * @param string $baseProjectDir
+     * @param string $modelDirPath
+     * @param string $wsdlPath
+     * @codeCoverageIgnore
+     */
+    public function __construct(AnnotationReader $annotationReader, string $baseProjectDir, string $modelDirPath, string $wsdlPath)
+    {
+        $this->annotationReader = $annotationReader;
+        $this->baseProjectDir = $baseProjectDir;
+        $this->modelDirPath = $modelDirPath;
+        $this->wsdlPath = $wsdlPath;
+    }
+
+    public function validate(): array
+    {
+        $missingFields = [];
+
+        foreach ($this->getAllClassAnnotations() as $annotation) {
+            $objectName = $annotation['object']->name;
+            $fieldNames = $this->getFieldNames($annotation['fields']);
+
+            if (!$objectName || !$fieldNames) {
+                continue;
+            }
+
+            foreach ($fieldNames as $fieldName) {
+                if ($this->hasField($objectName, $fieldName) === false) {
+                    $missingFields[$objectName][] = $fieldName;
+                }
+            }
+        }
+
+        return $missingFields;
+    }
+
+    private function getObject(string $objectName): ?string
+    {
+        preg_match("/<complexType name=\"$objectName\">([\w\W]*)<\/complexType>/", $this->getWsdlContents(), $matches);
+
+        return $matches[0] ?? null;
+    }
+
+    private function hasField(string $objectName, string $fieldName): bool
+    {
+        return strpos($this->getObject($objectName), sprintf('<element name="%s"', $fieldName));
```

Comment:
I'd suggest adding `!== false` or `>= 0` at the end as `strpos` returns int or bool. After this you're golden :+1: